annotations/views/project_views.py

```diff
@@ -8,60 +8,61 @@
 from django.core.paginator import Paginator, EmptyPage, PageNotAnInteger
 from django.core.exceptions import PermissionDenied
 from django.http import HttpResponse, HttpResponseRedirect
-from django.contrib.auth import login, authenticate
 from django.conf import settings
-from django.db.models import Q, Count
+from django.db.models import Q
 
-from annotations.models import TextCollection, RelationSet
+from annotations.models import TextCollection, VogonUser
 from annotations.forms import ProjectForm
 
+from django.contrib import messages
+from django.http import Http404
+
+from annotations.utils import get_ordering_metadata, get_user_project_stats, _annotate_project_counts
+
 @login_required
 def view_project(request, project_id):
     """
     Shows details about a specific project owned by the current user.
-
-    Parameters
-    ----------
-    request : `django.http.requests.HttpRequest`
-    project_id : int
-
-    Returns
-    ----------
-    :class:`django.http.response.HttpResponse`
     """
+    project = get_object_or_404(
+        _annotate_project_counts(TextCollection.objects),
+        pk=project_id
```

Comment:
I'm so confused by this, doesn't `get_object_or_404` expect a class, not a queryset? A

Reply:
hm, I think there needs to be a comment here, especially since it would be more obvious why you do that it here are many projects and you want to annotate them all, but this is only requesting one single one. It also doesn't help that it's the "TextCollection" class but the id is "project_id". So there should be an explanatory comment why this is used over just getting the counts directly from the requested project (the way it's written makes it a lot harder to understand what's going on).